godot/components/player/player.gd

```diff
@@ -2,28 +2,30 @@ extends CharacterBody2D
 
 @onready var animSprite = $AnimatedSprite2D;
 
-const playerSpeed := 500;
+const playerSpeed := 250;
+const jumpSpeed := 200;
+const gravity := 15;
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
 	animSprite.play("idle")
 
 func _physics_process(_delta):
-	var vertical_direction := Input.get_axis("up", "down")
 	var horizontal_direction := Input.get_axis("left", "right")
+		
+	if Input.is_action_pressed("up") and is_on_floor():
+		velocity.y -= jumpSpeed
+	else:
+		velocity.y += gravity
```

Comment:
it's beautiful.

Reply:
if you hit the corner of a block _just_ right, you can get some serious air. #speedrunstrats